README.adoc

```diff
@@ -22,40 +24,226 @@ Current
+        }
+        catch (InterruptedException interruptedException) {
+            throw new RuntimeException(interruptedException);
+        }
+
+        /*
+         * Send Hello, World! via rlp_01
+         */
+        new ExampleRelpClient(listenPort).send("Hello, World!");
+
+        /*
+         * Stop server
+         */
+        server.stop();
+
+        /*
+         * Wait for stop to complete
+         */
+        try {
+            serverThread.join();
+        }
+        catch (InterruptedException interruptedException) {
+            throw new RuntimeException(interruptedException);
+        }
+        System.out.println("server stopped at port <" + listenPort + ">");
+
+        /*
+         * Close the frameDelegate
+         */
+        try {
+            frameDelegate.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    /**
+     * ExampleRelpClient using rlp_01 for demonstration
+     */
+    private class ExampleRelpClient {
+        private final int port;
+        ExampleRelpClient(int port) {
+            this.port = port;
+        }
+
+        public void send(String record) {
+            RelpConnection relpConnection = new RelpConnection();
+            try {
+                relpConnection.connect("localhost", port);
+            }
+            catch (IOException | TimeoutException exception) {
+                throw new RuntimeException(exception);
+            }
+
+            RelpBatch relpBatch = new RelpBatch();
+            relpBatch.insert(record.getBytes(StandardCharsets.UTF_8));
+
+            while (!relpBatch.verifyTransactionAll()) {
+                relpBatch.retryAllFailed();
+                try {
+                    relpConnection.commit(relpBatch);
+                }
+                catch (IOException | TimeoutException exception) {
+                    throw new RuntimeException(exception);
+                }
+            }
+            try {
+                relpConnection.disconnect();
+            }
+            catch (IOException | TimeoutException exception) {
+                throw new RuntimeException(exception);
+            }
+            finally {
+                relpConnection.tearDown();
+            }
         }
     }
 }
 ----
 
+In case one requires a separate handler for each connection that does not need to be thread-safe create new FrameDelegate in the Supplier<FrameDelegate>:
+[source, java]
+----
+        Supplier<FrameDelegate> frameDelegateSupplier = () -> {
+            System.out.println("Providing frameDelegate for a connection");
+            return new DefaultFrameDelegate(frameContext -> System.out.println(frameContext.relpFrame().payload().toString()));
+        };
+----
+
+In case one requires deferred handler for command processing such implementation can be provided via passing custom RelpEvent implementation to DefaultFrameDelegate. See ReadmeDeferredTest.java for an example.
```

Comment:
> In case one requires deferred handler for command processing such implementation can be provided via passing custom RelpEvent implementation to DefaultFrameDelegate.

-> "If a deferred handler is required for command processing, pass custom `RelpEvent` implementation to `DefaultFrameDelegate`."